modules/configloader/include/configloader/BoardConf.hxx

```diff
@@ -32,23 +34,42 @@ namespace stdfs = std::filesystem;
 extern const char* board_conf_schema;
 
 struct BoardConf {
+    struct I2c {
+        std::uint16_t sda_pin;
+        std::uint16_t scl_pin;
+    };
     struct Uart {
         std::uint16_t rx_pin;
         std::uint16_t tx_pin;
+        std::uint16_t cts_pin; // Clear to send (in)
+        std::uint16_t rts_pin; // Request to send (out)
+    };
+    struct Spi {
+        std::uint16_t mosi_pin;
+        std::uint16_t miso_pin;
+        std::uint16_t clk_pin;
+        std::uint16_t cs_pin;
     };
 
     std::string name;
     std::string fqbn; // ArduinoCLI Fully Qualified Board Name
-    std::uint16_t digital_pin_count;
-    std::uint16_t analog_pin_count;
+    std::uint16_t pin_count;
 
-    std::vector<std::uint16_t> interruptable_digital_pins;
+    std::vector<std::uint16_t> digital_in_unable_pins;
+    std::vector<std::uint16_t> digital_out_unable_pins;
+    std::vector<std::uint16_t> analog_in_capable_pins;
+    std::vector<std::uint16_t> analog_out_capable_pins;
```

Comment:
Using unable alongside capable naming is a bit odd

Reply:
Reason I chose to go that route is so we need to write the least amount of pins in the config file.
Most boards have a large number of bidirectional digital pins, but a few number of analog-capable pins.